### ISSUE TITLE
Fix and enforce more warnings

### ISSUE THREAD

**Issue by jsichi** [jgrapht/jgrapht#1055]
```
 * JGraphT version: 1.5.1
```

See #1051 for details.  We want the build to always run clean with -Dmaven.compiler.showWarnings=true.  Before we can do that, we need to first fix the existing warnings, and then change the build to run with this setting by default.

**Comment by HannesWell** [comment]
My Eclipse shows me in total 369 warnings in the JGraphT library. Most of them are in tests, often they are issued due to unused elements or by the use of deprecated (jUnit) methods.
If you want I can fix those warnings that don't require major refactoring.

**Comment by jsichi** [comment]
Sorry for being imprecise; the intent of this issue is to fix the warnings that currently show up with this mvn command:

```
mvn -Dmaven.compiler.showWarnings=true -Dmaven.compiler.showDeprecation=true clean compile
```

(We run this as part of our release process to make sure we have removed all previously deprecated elements.)

For the Eclipse/test warnings, that would be a nice-to-have cleanup as a separate issue.

**Comment by jsichi** [comment]
Let's use a separate issue (I've edited the name for this one to be more specific).
